File: scripts/createNetwork.swift
// Script for creating a fully connected random network.
//
// To run...
//  $ swift createNetwork > <filename>
//
// *** Only will run on macOS *** 

import Darwin // for arc4random_uniform

// Configuration

let numInNodes = 64
let numOutNodes = 32
let hiddenLayers = [40, 34]

// Objects and functions

struct Node {
    private static var nextId = 0

    let id: Int

    init () {
        self.id = Node.nextId
        Node.nextId += 1
    }
}

class Conn {
    private static var __nextInnov = -1

    static var nextInnov: Int {
        __nextInnov += 1
        return __nextInnov
    }
}

func randomWeight () -> Double {
    return 2 * ((Double(arc4random_uniform(100)) / 100.0) - 0.5)
}

// Execution Start

var hiddenNodeCount = 0
for i in hiddenLayers {
    hiddenNodeCount += i
}
let totalNodeCount = numInNodes + numOutNodes + hiddenNodeCount

print("\(totalNodeCount) \(numInNodes) \(hiddenNodeCount) \(numOutNodes)")

var layers = [[Node]]()

// Add input nodes

layers.append([Node]())
for _ in 0..<numInNodes {
    layers[0].append(Node())
}

// Add hidden nodes
for i in 0..<hiddenLayers.count {
    layers.append([Node]())
    for _ in 0..<hiddenLayers[i] {
        layers[i + 1].append(Node())
    }
}

// Add output nodes
layers.append([Node]())
for _ in 0..<numOutNodes {
    layers[hiddenLayers.count + 1].append(Node())
}

// Create connections and print them out
for i in 0..<(layers.count - 1) {
    let baseLayer = layers[i]
    let targetLayer = layers[i + 1]
    for baseNode in baseLayer {
        for targetNode in targetLayer {
            let weight = randomWeight()
            let innov = Conn.nextInnov
            print("\(baseNode.id) \(targetNode.id) \(weight) \(innov) 1")
        }
    }
}

File: _delet this/scripts/createNetwork.py
from random import uniform

#
# Script for creating a fully connected, random network.
#
# There is no file manipluation in this script so the output
# must be feed into a file using the the unix pipeline and
# dos equivalent.
#
# Usage:
#   Configure the desired values in the configuration section.
#   Run command (Unix):
#       $ python createNetwork.py > <filename>
#   where <filename> is the file to output the data.
#
#   Run command (DOS): (probably...)
#       python createNetwork.py > <filename>
#   where <filename> is the file to output the data.
#

# Configuration
numInNodes = 3
numOutNodes = 3
hiddenLayers = [7, 5] # must have at least one element
# End of configuration

def randWeight (): 
    return uniform(-1, 1)

__nextNodeId = -1
__nextInnov = -1

def nextId ():
    global __nextNodeId
    __nextNodeId += 1
    return __nextNodeId

def nextInnov ():
    global __nextInnov
    __nextInnov += 1
    return __nextInnov

# Print header line
numHiddenNodes = 0
for layerCount in hiddenLayers:
    numHiddenNodes += layerCount

totalNodeCount = numInNodes + numOutNodes + numHiddenNodes

print totalNodeCount, numInNodes, numOutNodes, numHiddenNodes

# List of all layers in the network
nodes = []

# List of the input nodes as id values
inNodes = []

for i in range(0, numInNodes):
    inNodes.append(nextId())

nodes.append(inNodes)

# Append the hidden layers to the node list
for layer in hiddenLayers:
    l = []
    for i in range(0, layer):
        l.append(nextId())

    nodes.append(l)

# List of the output nodes as id values
outNodes = []

for i in range(0, numOutNodes):
    outNodes.append(nextId())

nodes.append(outNodes)

# Create connection and print them to stdout

for i in range(0, len(nodes) - 1):
    baseLayer = nodes[i]
    targetLayer = nodes[i + 1]

    for baseNode in baseLayer:
        for targetNode in targetLayer:
            # Printing of connections
            print baseNode, targetNode, nextInnov(), randWeight(), 1
